src/robot/libraries/Collections.py

```diff
@@ -698,60 +748,108 @@ def get_from_dictionary(self, dictionary, key, default=NOT_SET):
+                                      ignore_case=False):
         """Fails if ``key`` is not found from ``dictionary``.
 
         Use the ``msg`` argument to override the default error message.
+        
+        See section `Ignore Case` for more information about how to use ignore_case 
+        as an argument
         """
         self._validate_dictionary(dictionary)
+        should_ignore = self._is_ignore_case(ignore_case)
+        if should_ignore['key']:
+            dictionary=self._normalize(dictionary)
+            key=self._normalize(key)
         _verify_condition(key in dictionary,
                           f"Dictionary does not contain key '{key}'.",
                           msg)
 
-    def dictionary_should_not_contain_key(self, dictionary, key, msg=None):
+    def dictionary_should_not_contain_key(self, dictionary, key, msg=None,
+                                          ignore_case=False):
         """Fails if ``key`` is found from ``dictionary``.
 
         Use the ``msg`` argument to override the default error message.
+        
+        See section `Ignore Case` for more information about how to use ignore_case 
+        as an argument
         """
         self._validate_dictionary(dictionary)
+        should_ignore = self._is_ignore_case(ignore_case)
+        if should_ignore['key']:
+            dictionary=self._normalize(dictionary)
+            key=self._normalize(key)
         _verify_condition(key not in dictionary,
                           f"Dictionary contains key '{key}'.",
                           msg)
 
-    def dictionary_should_contain_item(self, dictionary, key, value, msg=None):
+    def dictionary_should_contain_item(self, dictionary, key, value, msg=None,
+                                       ignore_case=False):
         """An item of ``key`` / ``value`` must be found in a ``dictionary``.
 
         Value is converted to unicode for comparison.
 
         Use the ``msg`` argument to override the default error message.
+        
+        See section `Ignore Case` for more information about how to use ignore_case 
+        as an argument
         """
         self._validate_dictionary(dictionary)
-        self.dictionary_should_contain_key(dictionary, key, msg)
+        should_ignore = self._is_ignore_case(ignore_case)
+        self.dictionary_should_contain_key(dictionary, key, msg, should_ignore['key'])
+        if ignore_case is True or self._normalize(ignore_case)=="both":
+            dictionary = self._normalize(dictionary)
+            key=self._normalize(key)
+            value=self._normalize(value)
+        elif should_ignore['key']:
+            dictionary = {self._normalize(k): dictionary[k] for k in dictionary}
+            key=self._normalize(key)
+        elif should_ignore['value']:
+            dictionary = {k: self._normalize(dictionary[k]) for k in dictionary}
+            value=self._normalize(value)
         assert_equal(dictionary[key], value,
```

Comment:
This can be massively simplified by using the `Normalizer` class. There are also other places where it's useful but I won't mention it unless there's something special.